### ISSUE TITLE
Optional values in @llm.ai_callable cause ValueError

### ISSUE THREAD

**Issue by FiddlersCode** [livekit/agents#1185]
**How to replicate**
1. Set tool call params as below:
2. LLM sets this parameter to None

```
@llm.ai_callable(
    description="""
    "Returns a list of transactions during a specified period and filters...blah blah
    """
)
async def a_tool_call(
    self,
    start_date: Annotated[
        Optional[str],
        llm.TypeInfo(
            description="""
                The start date in the format %Y-%m-%d, 
                for example, 2020-09-01 for 1st September 2020.
                If you don't think this field is relevant for the question then set this parameter to None.
                """
        ),
    ] = None,
...more parameters
``` 

**Expected behaviour**
Tool call proceeds as usual with `None` set as a parameter

**Actual behaviour**
We receive the following error message and the tool call is now able to proceed
```
2024-11-29 17:34:13,600 - ERROR livekit.plugins.openai.realtime - failed to handle OpenAI S2S message
Traceback (most recent call last):
  File "/Users/paula.muldoon/dev/ai/ai-livekit-agent/.venv/lib/python3.11/site-packages/livekit/plugins/openai/realtime/realtime_model.py", line 1038, in _recv_task
    self._handle_response_output_item_done(data)
  File "/Users/paula.muldoon/dev/ai/ai-livekit-agent/.venv/lib/python3.11/site-packages/livekit/plugins/openai/realtime/realtime_model.py", line 1341, in _handle_response_output_item_done
    fnc_call_info = _oai_api.create_ai_function_info(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/paula.muldoon/dev/ai/ai-livekit-agent/.venv/lib/python3.11/site-packages/livekit/agents/llm/_oai_api.py", line 72, in create_ai_function_info
    sanitized_value = _sanitize_primitive(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/Users/paula.muldoon/dev/ai/ai-livekit-agent/.venv/lib/python3.11/site-packages/livekit/agents/llm/_oai_api.py", line 148, in _sanitize_primitive
    raise ValueError(f"expected str, got {type(value)}")

ValueError: expected str, got <class 'NoneType'>
```

I've managed to print out these lines from `livekit/agents/llm/_oai_api.py`
line 50:
```

         arg_value = parsed_arguments[arg_info.name]
        print(f"arg info: {arg_info.name}")
        print(f"arg value: {arg_value}")
        print(f"arg type: {arg_info.type}")
```
and the result is:
```
arg info: start_date
arg value: None
arg type: <class 'str'>
```
Versions
livekit = "0.18.1"
livekit-agents = "0.11.3"
livekit-plugins-openai = "0.10.7"
livekit-protocol = "0.7.0"

**Comment by jayeshp19** [comment]
Hi @FiddlersCode, Could you please try updating the `livekit-plugins-openai` package to version `0.10.9` and also ensure the `livekit-agents` package is updated to the latest version(0.12.0)? Once updated, give it another try and let me know if the issue persists.

**Comment by davidzhao** [comment]
fixed in #1211

**Comment by lomalan** [comment]
Hey @davidzhao I have the same issue on the livekit-agents = "0.12.8"

```
Value None is optional True
I'm sanitizing value to_account, None, <class 'str'>, ()
2025-01-30 16:03:47,060 - ERROR livekit.plugins.openai.realtime - failed to handle OpenAI S2S message
Traceback (most recent call last):
  File "/Users/anton.lomakin/Library/Caches/pypoetry/virtualenvs/ai-livekit-agent-jShpCBEX-py3.11/lib/python3.11/site-packages/livekit/plugins/openai/realtime/realtime_model.py", line 1195, in _recv_task
    self._handle_response_output_item_done(data)
  File "/Users/anton.lomakin/Library/Caches/pypoetry/virtualenvs/ai-livekit-agent-jShpCBEX-py3.11/lib/python3.11/site-packages/livekit/plugins/openai/realtime/realtime_model.py", line 1556, in _handle_response_output_item_done
    fnc_call_info = _create_ai_function_info(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/anton.lomakin/Library/Caches/pypoetry/virtualenvs/ai-livekit-agent-jShpCBEX-py3.11/lib/python3.11/site-packages/livekit/agents/llm/function_context.py", line 358, in _create_ai_function_info
    sanitized_value = _sanitize_primitive(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/Users/anton.lomakin/Library/Caches/pypoetry/virtualenvs/ai-livekit-agent-jShpCBEX-py3.11/lib/python3.11/site-packages/livekit/agents/llm/function_context.py", line 379, in _sanitize_primitive
    raise ValueError(f"expected str, got {type(value)}")
ValueError: expected str, got <class 'NoneType'>
```

Could you please help us with it?
